ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/GenerateMissingRecordFieldsFix.fs

```diff
@@ -101,37 +77,32 @@ type GenerateMissingRecordFieldsFix(recordExpr: IRecordExpr) =
 
         let generatedBindings = List<IRecordExprBinding>()
 
-        let indent =
-            match anchor with
-            | :? IRecordExprBinding -> anchor.Indent
-            | _ -> anchor.Indent + 1
+        let anchorBindingList =
+            match existingBindings.LastOrDefault() with
+            | null ->
+                let firstField = fieldsToAdd.First()
+                fieldsToAdd.Remove(firstField) |> ignore
+                let binding = elementFactory.CreateRecordExprBinding(firstField, generateSingleLine)
+                let bindingList = RecordExprBindingListNavigator.GetByExprBinding(binding)
+                let actualList = ModificationUtil.AddChildAfter(recordExpr.LeftBrace, bindingList)
+                generatedBindings.Add(actualList.ExprBindings.First())
+                actualList
+            | binding -> RecordExprBindingListNavigator.GetByExprBinding(binding)
 
         for name in fieldsToAdd do
-            if anchor :? IRecordExprBinding then
-                if generateSingleLine then
-                    anchor <- ModificationUtil.AddChildAfter(anchor, Whitespace())
-                else
-                    anchor <- ModificationUtil.AddChildAfter(anchor, NewLine(lineEnding))
-                    anchor <- ModificationUtil.AddChildAfter(anchor, Whitespace(indent))
-
             let binding = elementFactory.CreateRecordExprBinding(name, generateSingleLine)
-            anchor <- ModificationUtil.AddChildAfter(anchor, binding)
-            generatedBindings.Add(anchor :?> _)
-
-        let lastBinding = generatedBindings.Last()
+            generatedBindings.Add(ModificationUtil.AddChild(anchorBindingList, binding))
 
         if generateSingleLine then
+            let lastBinding = generatedBindings.Last()
             ModificationUtil.DeleteChild(lastBinding.Semicolon)
 
             for binding in existingBindings do
                 if binding.NextSibling :? IRecordExprBinding then
                     ModificationUtil.AddChildAfter(binding, Whitespace()) |> ignore
 
-            if recordExpr.LeftBrace.NextSibling :? IRecordExprBinding then
-                ModificationUtil.AddChildAfter(recordExpr.LeftBrace, Whitespace()) |> ignore
-
-        if lastBinding.NextSibling.GetTokenType() == FSharpTokenType.RBRACE then
-            ModificationUtil.AddChildAfter(lastBinding, Whitespace()) |> ignore
+        if recordExpr.LeftBrace.NextSibling :? IRecordExprBindingList then
+            ModificationUtil.AddChildAfter(recordExpr.LeftBrace, Whitespace()) |> ignore
```

Comment:
If there are at least two existing bindings, then formatter won't be called on a range that includes left brace, so we need to manually add this whitespace.

Reply:
OK, I see. I think we should be able to trigger formatter anyway, since it'd be great to fix other cases too, like:
```fsharp
{    ManySpacesBefore = true
     WouldBeGreatToHaveOneSpace = 123 }
```